### PR TITLE
Improve driver_updates logging

### PR DESCRIPTION
**Pull request by jkonecny12**
Debugging of the `driver_updates` script could be a pain. To improve that we need much more logs in the whole file in a way that it won't bother users during boot. To solve that we are adding debug logs and showing them on the console only in case that `inst.debug` or `rd.debug` is used.

### CODE REVIEW

**Comment by jkonecny12** [comment]
@jstodola what do you think, any ideas for improvements?

**Review by rvykydal — APPROVED** [review]
Looks good to me, thank you.

**Comment by jstodola** [comment]
@jkonecny12, I'm on my phone only for a while, so it's quite challenging to suggest any improvements. So, no useful feedback from my side :-)

**Comment by jkonecny12** [comment]
This change has time. I know that I will wait for you to have time, no problem with that.

File: dracut/driver_updates.py [hunk 1]
```diff
@@ -428,17 +430,21 @@ def load_drivers(moddict):
...
                       intfs_removed, net_intfs_unload)
 
     # Step 2: Update the depmod data and try to load the new module list
-    log.debug("load_drivers: %s", moddict.keys())
+    log.debug("updating depmod data")
     subprocess.call(["depmod", "-a"])
 
+    log.debug("load_drivers: %s", list(moddict.keys()))
     if moddict:
+        log.debug("insert modules %s", list(moddict.keys()))
```
[the remark this answers is not in the document]
Comment (jkonecny12) [reply dracut/driver_updates.py:447]: Fixed.

File: dracut/driver_updates.py [hunk 2]
```diff
@@ -853,6 +860,7 @@ def main(args):
     # or we need to re-trigger the block rules.
     if os.path.exists("/tmp/anaconda-dd-on-cdrom") and not os.path.exists("/dev/root"):
         cmd = ["udevadm", "trigger", "--action=change", "--subsystem-match=block"]
+        log.debug("trigger udevadm to mount cdrom with stage2 image")
```
[the remark this answers is not in the document]
Comment (jkonecny12) [reply dracut/driver_updates.py:863]: Fixed.

**Comment by jkonecny12** [comment]
**UPDATED:**
*Apply notes from @VladimirSlavik - thanks!
*Add new commit to move the file into the pep8.

File: dracut/driver_updates.py [hunk 3]
```diff
@@ -428,17 +458,21 @@ def load_drivers(moddict):
...
         if intfs_removed != net_intfs_unload:
             log.error("ERROR: removed %s interfaces are not expected interfaces for removal %s",
                       intfs_removed, net_intfs_unload)
 
     # Step 2: Update the depmod data and try to load the new module list
-    log.debug("load_drivers: %s", moddict.keys())
+    log.debug("updating depmod data")
     subprocess.call(["depmod", "-a"])
 
+    log.debug("load_drivers: %s", list(moddict.keys()))
```
Comment (poncovka) [inline dracut/driver_updates.py:473]: It looks like a leftover.

**Review by poncovka — APPROVED** [review]

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Comment by jkonecny12** [comment]
**UPDATED:**
Seems that my previous implementation did not worked as I expected. There were duplicate messages for DEBUG and INFO to the console.

So I changed the behavior a bit to always have a prefix `DD (<LEVEL>)`. It could feel a kind a weird so I'm thinking if I shouldn't remove that to when logging into console. What do you think?

anaconda-dd-debugging

**Comment by VladimirSlavik** [comment]
Do you mean removing only the prefixes in console? That sounds good.

**Comment by jkonecny12** [comment]
Yes, I can leave the prefixes in the syslog but can remove them in the console. It will be problem to distinguish between debug and info message but that shouldn't be problem thought.

**Comment by jkonecny12** [comment]
@jstodola any opinion about this?

**Comment by jkonecny12** [comment]
Switching to Fedora 38

**Comment by jkonecny12** [comment]
Updated the behavior.

Remove prefixes from the console prints when debug is on:
driver_updates_debug_without_prefix
mounting and unmounting is the debug output


Leave prefixes in the syslog:
driver_updates_no_debug_journal

**Comment by jkonecny12** [comment]
**UPDATED:**
* fix tests

**Review by poncovka — APPROVED** [review]

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fix an issue when `inst.debug` was last on the command line (the `\n` broke detection)

Tested and it seems that everything works as it should now.

**Pull request merged**